jetty-websocket/websocket-javax-client/src/main/java/org/eclipse/jetty/websocket/javax/client/internal/JavaxWebSocketClientContainer.java

```diff
@@ -271,4 +281,75 @@ private ClientEndpointConfig getAnnotatedConfig(Object endpoint) throws Deployme
 
         return new AnnotatedClientEndpointConfig(anno);
     }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        doClientStart();
+        super.doStart();
+    }
+
+    protected void doClientStart()
+    {
+        // If we are running in Jetty register shutdown with the ContextHandler.
+        // TODO: add test mode to disable this.
+        if (shutdownWithContextHandler(this))
```

Comment:
This is still needed, because the SCI shutdown mechanism won't work when running with embedded Jetty as we wont have a different classloader for every context, so the static will be the same for every context.